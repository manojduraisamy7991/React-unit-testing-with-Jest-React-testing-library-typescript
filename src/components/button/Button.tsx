import './button.css';

interface FormatDateProps {
  label: String;
}

function Button({ label }: FormatDateProps): JSX.Element {
  return (
    <div data-testid="button" className="button-style">
      hi{label.toLocaleString()}
    </div>
  );
}

export default Button;
