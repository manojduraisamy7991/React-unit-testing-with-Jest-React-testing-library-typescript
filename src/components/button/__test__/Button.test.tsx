/* eslint-disable testing-library/prefer-screen-queries */
import ReactDOM from 'react-dom';
import '@testing-library/jest-dom/extend-expect';
import { render, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Button from '../Button';

afterEach(cleanup);

it('render without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Button label="click"></Button>, div);
});

it('renders button correctly', () => {
  const { getByTestId } = render(<Button label="Click"></Button>);
  expect(getByTestId('button')).toHaveTextContent('Click');
});

it('renders button correctly label', () => {
  const { getByTestId } = render(<Button label="save "></Button>);
  expect(getByTestId('button')).toHaveTextContent('save');
});

it('matches snapshot1', () => {
  const tree = renderer.create(<Button label="save"></Button>).toJSON();
  expect(tree).toMatchSnapshot();
});

it('matches snapshot2', () => {
  const tree = renderer.create(<Button label="Click"></Button>).toJSON();
  expect(tree).toMatchSnapshot();
});
