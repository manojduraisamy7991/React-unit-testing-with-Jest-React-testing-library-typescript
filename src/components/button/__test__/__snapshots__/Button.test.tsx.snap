// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot1 1`] = `
<div
  className="button-style"
  data-testid="button"
>
  hi
  save
</div>
`;

exports[`matches snapshot2 1`] = `
<div
  className="button-style"
  data-testid="button"
>
  hi
  Click
</div>
`;
